src/main/java/ua/samoplavskaya/hw_dao/dao/impl/JDBCDao.java

```diff
@@ -0,0 +1,108 @@
+package ua.samoplavskaya.hw_dao.dao.impl;
+
+import ua.samoplavskaya.hw_dao.dao.Dao;
+import ua.samoplavskaya.hw_dao.entity.Person;
+import ua.samoplavskaya.hw_dao.util.JDBCConnectionUtil;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class JDBCDao implements Dao<Person> {
+    private static final JDBCDao INSTANCE = new JDBCDao();
+
+    private JDBCDao() {
+    }
+
+    public static JDBCDao getInstance() {
+        return INSTANCE;
+    }
+
+    private static final String SAVE_SQL = "INSERT INTO persons(first_name,last_name,email) VALUES (?,?,?);";
+    private static final String FIND_SQL = "SELECT * FROM persons WHERE id= ?;";
+    private static final String FIND_ALL = "SELECT * FROM persons ORDER BY id;";
+    private static final String UPDATE = "UPDATE persons SET first_name=?,last_name=?,email=? WHERE id=?;";
+    private static final String DELETE = "DELETE FROM persons WHERE id=?;";
+
+    @Override
+    public Optional<Person> findByID(long id) {
+        try (Connection connection = JDBCConnectionUtil.get()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(FIND_SQL);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Person person = new Person(
+                        resultSet.getLong("id"),
+                        resultSet.getString("first_name"),
+                        resultSet.getString("last_name"),
+                        resultSet.getString("email")
+                );
```

Comment:
N: In findAll you have the same part of code so you can move it as separate method so as not to duplicate yourself.